code/src/constants/requests.ts

```diff
@@ -12,5 +14,11 @@ export const URLS = {
   orders: {
     get: "/v1/orders/user",
     post: "/v1/orders"
+  },
+  cart: {
+    post: ({ userId, productId }: CartManagementParams) =>
+      `/v1/users/${userId}/cart/${productId}`,
+    delete: ({ userId, productId }: CartManagementParams) =>
+      `/v1/users/${userId}/cart/items/${productId}`
```

Comment:
I suggest using `satistfies` keyword for stricter type limitations, because we want object, which keys are also object with name as a key and string or function that returns a string as a value:

I suggest following approach:

```jsx
type UrlPath = string | ((...tokens: string[]) => string);

export const URLS = {
  auth: {
    signUp: "/auth/sign-up",
    signIn: "/auth/sign-in"
  },
  products: {
    get: "/v1/products",
    post: "/v1/products",
    put: "/v1/products",
    delete: "/v1/products"
  },
  orders: {
    get: "/v1/orders/user",
    post: (id) => /v1/users/${id}/orders
  }
} as const satisfies Record<string, Record<string, UrlPath>>;
```

Maybe you can adjust it to make function arguments more generic, not `string[]` as I demonstrated

Reply:
Maybe would be good to create an issue to add `satisfies` keyword for our constant data like router paths, URLs, etc. For now I couldn't add it because of unexpected errors. Code like this throws error for example:
```
type UrlFunctionParam = number | string | Record<string, string | number>;
type UrlFunction = (param: UrlFunctionParam) => string;
type UrlPath = string | UrlFunction;

export const URLS = {
  auth: {
    signUp: "/auth/sign-up",
    signIn: "/auth/sign-in"
  },
  products: {
    get: "/v1/products",
    post: "/v1/products",
    put: "/v1/products",
    delete: "/v1/products"
  },
  orders: {
    get: "/v1/orders/user",
    post: "/v1/orders"
  },
  cart: {
    post: ({ userId, productId }: CartManagementParams) =>
      `/v1/users/${userId}/cart/${productId}`,
    delete: ({ userId, productId }: CartManagementParams) =>
      `/v1/users/${userId}/cart/items/${productId}`
  }
} as const satisfies Record<string, Record<string, UrlPath>>;
```